arangod/Replication2/StateMachines/Document/ReplicatedOperation.h

```diff
@@ -0,0 +1,267 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2014-2023 ArangoDB GmbH, Cologne, Germany
+/// Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Alexandru Petenchea
+////////////////////////////////////////////////////////////////////////////////
+
+#pragma once
+
+#include "Cluster/ClusterTypes.h"
+#include "Inspection/Format.h"
+#include "Inspection/Status.h"
+#include "Inspection/Types.h"
+#include "VocBase/Identifiers/TransactionId.h"
+
+#include <variant>
+
+namespace arangodb::replication2::replicated_state::document {
+
+/*
+ * When a log entry is received, the ReplicatedOperation tells the state machine
+ * what it has to do (i.e. start a transaction, drop a shard, ...).
+ */
+struct ReplicatedOperation {
```

Comment:
This comment holds for all of the following file.

Do we need these elements exactly where they are serialized/deserialzed as well?
Or are they passed around, and only serialized/deserialized in fewer files?

If the second is true, we should have a follow up task to split "Inspectors" from "Structs" into two separate header files.
The inspector has high impact on compile time, that we can avoid for files just using the structs, but have nothing to do with their outside representation

Reply:
That's a good observation, I have split them into separate files!